src/containers/settings/EditSettingsScreen.tsx

```diff
@@ -259,6 +259,10 @@ const messages = defineMessages({
     id: 'settings.app.form.alwaysShowWorkspaces',
     defaultMessage: 'Always show workspace drawer',
   },
+  useWorkspaceDrawerIconStyle: {
+    id: 'settings.app.form.useWorkspaceDrawerIconStyle',
+    defaultMessage: 'Use workspace drawer icon style',
```

Comment:
So, I was suggesting to have the user choose whether they want the name or the first character of the workspace name AND/OR the icon. Could the feature be imagined like that?

Reply:
If I understand correctly. 

The letter is currently a fallback for workspaces that don't have an icon.

The setting should be a select with theses choices :

- text  (current display)
- (letter OR icon) AND text
- letter AND icon (if icon configured)
- letter OR icon

It means, 3 different workspace drawer sizes 

- the current (300px by default)
- icon OR letter only (75px)
- letter + icon (let's say 150px)

Reply:
(this may be not that easy to implement)

But should the workspace drawer bar be the same size of the service bar ? 
I means use the serviceRibbonWidth parameter for both workspace and service bar

Reply:
@vraravam Can you confirm that I have understood correctly ?
If yes, I will start the dev this week   😀

Reply:
we don't need to complicate so much. only 2 options that i was suggesting:
1. workspace name (current)
2. icon or first character of workspace name (if the user wants to use a smaller-width option for the workspace drawer)

Reply:
Okey perfect.😀

The current state of the code pushed for this PR match what you are describing. 
I've added an option to choose between these two modes. 

<img width="335" alt="image" src="https://github.com/ferdium/ferdium-app/assets/25098751/e8ac6a88-c4b9-4616-abfb-8d07fd7e8aba">



I just need to permanently store the icon field. 
Can you point me which files should I change to make this new workspace field persistent ?
I may need to do it for the local and distant server if I understand correctly.

Reply:
To permanently save this field you would probably need to sync with the server (internal and external) if I'm correct.

I'm not recalling if the server is saving workspace data as a plain JSON or not (only by checking the endpoint we can be sure of that). If that is not the case, you would need to touch both the server and the internal-server... but with the ongoing Adonis update this could be a hassle :/ I can help on that tho.

Reply:
Also, can we make this option the default one? I like this new look (even though I'm not using workspaces).

Great work!!

Reply:
yes, you will need to sync to the internal and external servers. The db schema is the same across both - and there's a table called `workspaces` to store this info.
<img width="704" alt="Screenshot 2023-09-05 at 8 18 24 AM" src="https://github.com/ferdium/ferdium-app/assets/69629/c3581cec-6f70-42c9-9db7-a37f7abfe67d">

As seen in the above pic (taken from my internal server), you can either add a new column for the `useIcon` or you can add it into the `data` column which holds JSONB datatype. (I personally prefer option 1.

Reply:
> yes, you will need to sync to the internal and external servers. The db schema is the same across both - and there's a table called `workspaces` to store this info. <img alt="Screenshot 2023-09-05 at 8 18 24 AM" width="704" src="https://user-images.githubusercontent.com/69629/265572474-c3581cec-6f70-42c9-9db7-a37f7abfe67d.png">
> 
> As seen in the above pic (taken from my internal server), you can either add a new column for the `useIcon` or you can add it into the `data` column which holds JSONB datatype. (I personally prefer option 1.

Even though I would prefer option 1 (just like you) I would advise into adding it to "data" column (but first lets be sure what type of data is being retrieved there) so that it doesn't make the migration to the new Adonis's even more complicated (as we would to update both the code in the current ferdium-server, internal server and on the current adonis update PR)

Reply:
As suggested, i added the iconUrl field in `data`

WIP related PR on the server : https://github.com/ferdium/ferdium-server/pull/76